### ISSUE TITLE
understanding of 

### ISSUE THREAD

**Issue by as142** [mikunit567/GAE#1]
Hi there, When I was learning and trying to understand the function "ValidateXsrfToken" in GAE/google_appengine/google/appengine/ext/datastore_admin/utils.py, I was confused about the description of its delivered function. My understanding of this code snippet is "validate a given xsrf token by retrieving it". Do I understand that right?

**Comment by mikunit567** [comment]
> Hi there, When I was learning and trying to understand the function "ValidateXsrfToken" in GAE/google_appengine/google/appengine/ext/datastore_admin/utils.py, I was confused about the description of its delivered function. My understanding of this code snippet is "validate a given xsrf token by retrieving it". Do I understand that right?

Yes, this is correct. Validate a retrieved XSRF from the memory cache and then with the token perform an associated action.

**Comment by as142** [comment]
Thank you so much for your response.
